jenkins/artifact-signer/Jenkinsfile

```diff
@@ -0,0 +1,72 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters{
+        string(
+                name: 'ARTIFACT_PATH',
+                trim: true,
+                description: 'Path to the artifact or the artifact directory on s3'
+        )
+        choice(
+                choices: ['linux'],
+                name: 'DISTRIBUTION_PLATFORM',
+                description: 'What platform is this distribution build for?'
+        )
+        choice(
+                choices: ['.sig'],
+                name: 'SIGNATURE_TYPE',
+                description: 'What is signature file type?'
+        )
+    }
+    stages {
+        stage('sign') {
+            steps {
+                script {
+                    if (ARTIFACT_PATH == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('ARTIFACT_PATH is not set')
+                    }
+                    ARTIFACT_PATH = ARTIFACT_PATH.replaceAll('/$', "")
+                    set_up()
+                    signArtifacts(
+                            artifactPath: "$WORKSPACE/artifacts/${ARTIFACT_PATH}",
+                            signatureType: SIGNATURE_TYPE,
+                            distributionPlatform: DISTRIBUTION_PLATFORM
+                    )
+                    upload_artifacts()
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                    }
+                }
+            }
+        }
+    }
+}
+
+void upload_artifacts(){
```

Comment:
Groovy convention is `camelCase`, so `uploadArtifacts()`. And to make this self-explanatory I would rename it to `uploadSignedArtifacts`.

Note though that these functions are not namespaced and I think may conflict with globals in `vars`. I don't know how Jenkins loads these things to be sure. Generally, I am not sure it's worth extracting functions that aren't called multiple times. But if we upload with this convention from `path` to `path/signed`, then extract this into a shared function in `vars`.

Reply:
I created a function to upload to s3 and updated the code in other files to use the function instead.